File: Sprint 2/Doctor.java
import java.util.*;
import java.time.*;

public class Doctor{
    private String firstName;
    private String lastName;
    private List<Patient> patients;
    private List<Regimen> regimens;

    Doctor(String firstName, String lastName){
        this.firstName = firstName;
        this.lastName = lastName;
        this.patients = new ArrayList<>(); 
        this.regimens = new ArrayList<>();
    }

    public void addPatient(Patient patient){
        patients.add(patient);
    }

    public void createPatient(String firstName, String lastName, int year, int month, int day){
        Patient patient = new Patient(firstName, lastName, year, month, day);
        addPatient(patient);
    }

    public void createRegimen(Patient patient, String medication, String direction){
        Regimen regimen = new Regimen(patient, this.Doctor, medication, direction);
        addRegimen(regimen);
        patient.addRegimen(regimen);
    }

    public void addRegimen(Regimen regimen){
        regimens.add(regimen);
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public List<Patient> getPatients() {
        return patients;
    }

    public Patient getPatient(int i) {
        return patients.get(i);
    }

    public List<Regimen> getRegimens() {
        return regimens;
    }

    public Regimen getRegimen(int i) {
        return regimens.get(i);
    }
}

File: Sprint_3/Doctor.js
import { Patient } from './Patient.js';
import { Regimen } from './Regimen.js';

export class Doctor {
    constructor(firstName, lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.patients = [];
        this.regimens = [];
    }

    addPatient(patient) {
        this.patients.push(patient);
    }

    createPatient(firstName, lastName, year, month, day) {
        const patient = new Patient(firstName, lastName, year, month, day);
        this.addPatient(patient);
        return patient;
    }

    createRegimen(patient, medication, direction) {
        const regimen = new Regimen(patient, this, medication, direction);
        this.addRegimen(regimen);
        patient.addRegimen(regimen);
        return regimen;
    }

    addRegimen(regimen) {
        this.regimens.push(regimen);
    }

    getFirstName() {
        return this.firstName;
    }

    getLastName() {
        return this.lastName;
    }

    getPatients() {
        return this.patients;
    }

    getPatient(i) {
        return this.patients[i];
    }

    getRegimens() {
        return this.regimens;
    }

    getRegimen(i) {
        return this.regimens[i];
    }
}